lib/library_version_analysis/mobile.rb

```diff
@@ -152,15 +154,22 @@ def build_transitive_mapping(parsed_results)
       # ├ ─ ┬    │ ├ ─ ─   │ │ └ ─ ─ These are the symbols used, keep here for now
       parent = "undefined"
 
+      last_parent = false
       results.each_line do |line|
         scan_result = line.scan(/.* (.*)@(.*)/)
         next if scan_result.nil? || scan_result.empty?
 
         name = scan_result[0][0]
+
         current_version = scan_result[0][1]
         parsed_results[name].current_version = current_version unless parsed_results[name].nil?
 
-        parent = name if !name.nil? && (line.count("│").zero? || line[0] == "├")
+        if line.start_with?("└")
```

Comment:
Some extra code was needed to handle an edge case in parsing the list of npm modules